### ISSUE TITLE
DCHECK failure with virtual keyboard

### ISSUE THREAD

**Issue by simonhong** [brave/brave-browser#15982]
With virtual keyboard on, I can get this failure. cc @iefremov 
No specific repro step. Just used browser and tried to type characters with virtual keyboard.
```
[16184:259:0520/090742.766048:FATAL:ax_position.h(3384)] Check failed: (result == 0 && slow_result == 0) || (result < 0 && slow_result < 0) || (result > 0 && slow_result > 0). 2 vs. 0
0   libbase.dylib                       0x00000001118090b9 base::debug::CollectStackTrace(void**, unsigned long) + 9
1   libbase.dylib                       0x00000001116de613 base::debug::StackTrace::StackTrace() + 19
2   libbase.dylib                       0x0000000111703aff logging::LogMessage::~LogMessage() + 175
3   libbase.dylib                       0x00000001117049de logging::LogMessage::~LogMessage() + 14
4   libcontent.dylib                    0x000000011481273e ui::AXPosition<ui::AXNodePosition, ui::AXNode>::CompareTo(ui::AXPosition<ui::AXNodePosition, ui::AXNode> const&) const + 1518
5   libcontent.dylib                    0x0000000115179980 ui::AXRange<ui::AXPosition<ui::AXNodePosition, ui::AXNode> >::GetText(ui::AXTextConcatenationBehavior, int, bool, unsigned long*) const + 208
6   libcontent.dylib                    0x000000011517ff11 (anonymous namespace)::GetTextForTextMarkerRange(objc_object*) + 97
7   libcontent.dylib                    0x000000011517e913 -[BrowserAccessibilityCocoa accessibilityAttributeValue:forParameter:] + 5971
8   AppKit                              0x00007fff23477835 ___NSAccessibilityEntryPointValueForAttributeWithParameter_block_invoke.840 + 495
9   AppKit                              0x00007fff23472aae NSAccessibilityPerformEntryPointObject + 16
10  AppKit                              0x00007fff234742ef NSAccessibilityEntryPointValueForAttributeWithParameter + 172
11  AppKit                              0x00007fff2326c76d CopyParameterizedAttributeValue + 460
12  HIServices                          0x00007fff258d0897 _AXXMIGCopyParameterizedAttributeValue + 439
13  HIServices                          0x00007fff258f33ba _XCopyParameterizedAttributeValue + 608
14  HIServices                          0x00007fff258b093c mshMIGPerform + 182
15  CoreFoundation                      0x00007fff20638214 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 41
16  CoreFoundation                      0x00007fff206380f5 __CFRunLoopDoSource1 + 619
17  CoreFoundation                      0x00007fff2063677f __CFRunLoopRun + 2400
18  CoreFoundation                      0x00007fff2063575c CFRunLoopRunSpecific + 563
19  HIToolbox                           0x00007fff28857203 RunCurrentEventLoopInMode + 292
20  HIToolbox                           0x00007fff28856f65 ReceiveNextEventCommon + 587
21  HIToolbox                           0x00007fff28856d03 _BlockUntilNextEventMatchingListInModeWithFilter + 70
22  AppKit                              0x00007fff22e31b32 _DPSNextEvent + 864
23  AppKit                              0x00007fff22e30305 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1364
24  libchrome_dll.dylib                 0x00000001079d8cd0 __71-[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:]_block_invoke + 64
25  libbase.dylib                       0x0000000111821702 base::mac::CallWithEHFrame(void () block_pointer) + 10
26  libchrome_dll.dylib                 0x00000001079d8c09 -[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:] + 153
27  AppKit                              0x00007fff22e22679 -[NSApplication run] + 586
28  libbase.dylib                       0x000000011182e54c base::MessagePumpNSApplication::DoRun(base::MessagePump::Delegate*) + 348
29  libbase.dylib                       0x000000011182c9cc base::MessagePumpCFRunLoopBase::Run(base::MessagePump::Delegate*) + 140
30  libbase.dylib                       0x00000001117b59ab base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta) + 651
31  libbase.dylib                       0x000000011175bd7a base::RunLoop::Run(base::Location const&) + 890
32  libcontent.dylib                    0x000000011496b72f content::BrowserMainLoop::RunMainMessageLoop() + 111
33  libcontent.dylib                    0x000000011496d442 content::BrowserMainRunnerImpl::Run() + 82
34  libcontent.dylib                    0x0000000114968b36 content::BrowserMain(content::MainFunctionParams const&) + 230
35  libcontent.dylib                    0x00000001155e7b1f content::ContentMainRunnerImpl::RunBrowser(content::MainFunctionParams&, bool) + 1263
36  libcontent.dylib                    0x00000001155e75c2 content::ContentMainRunnerImpl::Run(bool) + 514
37  libcontent.dylib                    0x00000001155e6387 content::RunContentProcess(content::ContentMainParams const&, content::ContentMainRunner*) + 2631
38  libcontent.dylib                    0x00000001155e645d content::ContentMain(content::ContentMainParams const&) + 29
39  libchrome_dll.dylib                 0x00000001060ab0e9 ChromeMain + 281
40  Brave Browser Development           0x0000000105ed5caf main + 287
41  libdyld.dylib                       0x00007fff20559f3d start + 1
42  ???                                 0x0000000000000004 0x0 + 4
```
